### ISSUE TITLE
[UX] Create site preview functionality for use with theme/style settings

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#1960]
Re: 
https://github.com/backdrop/backdrop-issues/issues/1905
https://github.com/backdrop/backdrop-issues/issues/1949
https://github.com/backdrop/backdrop-issues/issues/1935

Could look like this:
![image](https://cloud.githubusercontent.com/assets/5607236/15598032/41d47f74-23a4-11e6-8758-46fd25e31c09.png)

Or it could be more like some of the competitor UI's:
### Squarespace

![screenshot 2016-05-26 16 09 59](https://cloud.githubusercontent.com/assets/5607236/15588561/8c93109a-235c-11e6-8e04-e2e278b1f721.png)
![screenshot 2016-05-26 16 10 04](https://cloud.githubusercontent.com/assets/5607236/15588564/8e8d3dda-235c-11e6-9967-5b656170a160.png)
![screenshot 2016-05-26 16 10 07](https://cloud.githubusercontent.com/assets/5607236/15588573/94259a3a-235c-11e6-8065-34f641ac77b8.png)
#### Color chooser

![screenshot 2016-05-26 16 11 27](https://cloud.githubusercontent.com/assets/5607236/15588574/95dc380c-235c-11e6-8981-cf7a8b0e7d7d.png)
#### Font chooser

![screenshot 2016-05-26 16 11 33](https://cloud.githubusercontent.com/assets/5607236/15588575/9777744c-235c-11e6-9f7a-5639d227a5f2.png)
### Tumblr

![screenshot 2016-05-26 16 08 26](https://cloud.githubusercontent.com/assets/5607236/15588585/9c3dfbe0-235c-11e6-900c-937d3f3258bf.png)
![screenshot 2016-05-26 16 08 32](https://cloud.githubusercontent.com/assets/5607236/15588588/9e64b97c-235c-11e6-84bd-05614b61cd86.png)
#### Color chooser

![screenshot 2016-05-26 16 11 14](https://cloud.githubusercontent.com/assets/5607236/15588591/a046783e-235c-11e6-9f98-ac55b8e9f228.png)

**Comment by wesruv** [comment]
TL;DR - I think this preview functionality should accomodate a slinky sidebar-o-settings, and the zoomed out page style site previews. The form builder can choose the one most appropriate for their admin interface.

I'm not a fan of the skinny sidebar-o-settings for us because I think that will get ugly.
* we will have core and contrib adding to this settings UI (maybe even modules adding onto theme settings? web font, image, or other appearance type modules?)
* for some settings interfaces it's more valuable to have room for descriptive labels and field hints than it is to have a 100% scale preview
* we have more functionality than these competitors, some theme settings may not fit well

My crystal ball foresees crap UI if we try to force any previewable settings form into that small of a space like that.

An iframe preview would be easy to transform down to a size that fits on the page, or up to 100% (for zoom functionality). That'd allow us to do page refreshes without interrupting the admin experience, instead of needing to have JS written to dynamically swap out swaths of markup of that may be effected by settings changes.

See http://demo.patternlab.io/, the entire page is in an iframe, which allows them to preview breakpoints, see top right: 
![image](https://cloud.githubusercontent.com/assets/5607236/15831300/ec609628-2bea-11e6-9d5b-5a5fc11ea9bf.png)

I especially enjoy disco.

**Comment by olafgrabienski** [comment]
The nice viewport resizer is "ish.", developed by Brad Frost:

- https://github.com/bradfrost/ish.
- http://bradfrost.com/demo/ish/

**Comment by wesruv** [comment]
Oh, nice :) might steal that

**Comment by quicksketch** [comment]
I think the main component of this was solved in https://github.com/backdrop/backdrop-issues/issues/3186. Although it doesn't match @wesruv's design exactly, it does add the primary feature noted here, which is a live preview of the real site via an iframe on the color settings form. If there are other parts of this design we would like to utilize, let's make a separate issue for those changes.
